server/container/guice/jpa-common-guice/src/test/java/org/apache/james/TestJPAConfigurationModuleWithSqlValidation.java

```diff
@@ -19,32 +19,85 @@
-    private static final String VALIDATION_SQL_QUERY = "VALUES 1";
+        @Override
+        protected void configure() {
+            setupAuthenticationOnDerby();
+        }
 
-    @Override
-    protected void configure() {
+        @Provides
+        @Singleton
+        JPAConfiguration provideConfiguration() {
+            return jpaConfigurationBuilder()
+                    .username(DATABASE_USERNAME)
+                    .password(DATABASE_PASSWORD)
+                    .build();
+        }
+
+        private void setupAuthenticationOnDerby() {
+            try {
+                Connection conn = DriverManager.getConnection(JDBC_EMBEDDED_URL, DATABASE_USERNAME, DATABASE_PASSWORD);
+                Statement s = conn.createStatement();
+                // Setting and Confirming requireAuthentication
+                s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY('derby.connection.requireAuthentication', 'true')");
+
+                // Setting authentication scheme and username password to Derby
+                s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY('derby.authentication.provider', 'BUILTIN')");
+                s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY('derby.user." + DATABASE_USERNAME + "', '" + DATABASE_PASSWORD + "')");
+                s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY('derby.database.propertiesOnly', 'true')");
+
+                // Setting default connection mode to no access to restrict accesses without authentication information
+                s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY('derby.database.defaultConnectionMode', 'noAccess')");
+                s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY('derby.database.fullAccessUsers', 'james')");
+                s.executeUpdate("CALL SYSCS_UTIL.SYSCS_SET_DATABASE_PROPERTY('derby.database.propertiesOnly', 'false')");
+
+                s.close();
+                conn.close();
+            } catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        }
     }
 
-    @Provides
-    @Singleton
-    JPAConfiguration provideConfiguration() {
+    static JPAConfiguration.Builder jpaConfigurationBuilder() {
         return JPAConfiguration.builder()
                 .driverName(JDBC_EMBEDDED_DRIVER)
                 .driverURL(JDBC_EMBEDDED_URL)
                 .testOnBorrow(true)
                 .validationQueryTimeoutSec(2)
-                .validationQuery(VALIDATION_SQL_QUERY)
-                .build();
+                .validationQuery(VALIDATION_SQL_QUERY);
     }
+
+    String DATABASE_USERNAME = "james";
+    String DATABASE_PASSWORD = "james-secret";
+    String JDBC_EMBEDDED_URL = "jdbc:derby:memory:mailboxintegration;create=true";
+    String JDBC_EMBEDDED_DRIVER = org.apache.derby.jdbc.EmbeddedDriver.class.getName();
+    String VALIDATION_SQL_QUERY = "VALUES 1";
```

Comment:
Why are those constants not `private static final` anymore and not on top of the class?

Reply:
not ` private static final` -> because it is an interface so these qualifier are not required and would lead to inteliJ warnings.

Your remark regarding order is true: static constants should come before static methods ( but after static classes)

Reply:
Missed that it was an interface, my bad !